src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphTextOutputFormatterCallback.java

```diff
@@ -338,8 +361,27 @@ private void writeAction(ActionAnalysisMetadata action, PrintStream printStream)
     }
 
     stringBuilder.append('\n');
+  }
 
-    printStream.write(stringBuilder.toString().getBytes(UTF_8));
+  private void writeCommand(ActionAnalysisMetadata action, StringBuilder stringBuilder)
+      throws IOException, CommandLineExpansionException, InterruptedException, EvalException {
+    if (!(action instanceof CommandAction)) {
+      return;
+    }
+
+    boolean first = true;
+    for (String arg :
+        ((CommandAction) action)
+            .getArguments().stream()
+                .map(a -> internalToEscapedUnicode(a))
+                .collect(toImmutableList())) {
+      if (!first) {
+        stringBuilder.append(' ');
+      }
+      ScriptUtil.emitCommandElement(stringBuilder, arg, first);
```

Comment:
nit: add inline comments for the method call. E.g. 
```
 ScriptUtil.emitCommandElement(/* message= */ stringBuilder, /* commandElement= */ arg, /* isBinary= */ first);
```

Reply:
Done